pages/index.js

```diff
@@ -31,6 +35,8 @@ export default function HomePage({
         randomSurprise={randomSurprise}
         setRandomSurprise={setRandomSurprise}
         handleSurprise={handleSurprise}
+        isUploading={isUploading}
+        setIsUploading={setIsUploading}
```

Comment:
As mentioned before: Try to avoid  passing state setter functions as props. It's better to encapsulate them within handler functions.